src/Knp/Menu/MenuItem.php

```diff
@@ -563,23 +564,23 @@ public function actsLikeLast()
     /**
      * Implements Countable
      */
-    public function count()
+    public function count(): int
     {
-        return count($this->children);
+        return \count($this->children);
     }
 
     /**
      * Implements IteratorAggregate
      */
-    public function getIterator()
+    public function getIterator(): \ArrayIterator
```

Comment:
```suggestion
    public function getIterator(): \Traversable
```

Let's match the interface here. `ArrayIterator` is an implementation detail of this class that shouldn't be exposed.

Reply:
But `ArrayIterator` is not just an implementation of `Traversable`. Instead, it implements  `ArrayAccess` , `SeekableIterator` , `Countable` , `Serializable`

Reply:
It is an implementation of traversable. (through the seekable iterator)

Reply:
`ArrayIterator` ➡️ `SeekableIterator` ➡️ `Iterator` ➡️ `Traversable`

Reply:
What I'm saying here is that it's not **only** an implementation of Traversable, so it has methods that are not present in Traversable interface

Reply:
Right, it also has a lot of stuff we don't need. Nobody should call `getIterator()` directly. The method allows us to use this class in `foreach` loops. And for that purpose, `Traversable` is sufficient. If you promise an `ArrayIterator` you cannot change the iterator implementation later without breaking BC. Please use `Traversable` as return type.